libs/@blockprotocol/type-system/crate/src/ontology/entity_type/mod.rs

```diff
@@ -7,20 +7,23 @@ mod wasm;
 use std::collections::{HashMap, HashSet};
 
 pub use error::ParseEntityTypeError;
+use serde::{Deserialize, Serialize};
 
 use crate::{
+    ontology::entity_type::error::MergeEntityTypeError,
     url::{BaseUrl, VersionedUrl},
     AllOf, Links, MaybeOrderedArray, Object, OneOf, PropertyTypeReference, ValidateUrl,
     ValidationError, ValueOrArray,
 };
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(try_from = "raw::EntityType", into = "raw::EntityType")]
 pub struct EntityType {
     id: VersionedUrl,
     title: String,
     description: Option<String>,
     property_object: Object<ValueOrArray<PropertyTypeReference>>,
-    inherits_from: AllOf<EntityTypeReference>,
+    pub inherits_from: AllOf<EntityTypeReference>,
```

Comment:
specific reason why only `inherits_from` is pub?

Reply:
I mentioned this in the description: it’s the only field which is required to be pub for our downstream usage. I'm planning on making them all public.